### DISCUSSION TITLE
Migration to 1.4 - in operator

### DISCUSSION THREAD

**Discussion by johaven** [sqlalchemy/sqlalchemy#7096] [Usage Questions] [answered]
Hi ! I would like an example to migrate this kind of query:

```python
# get_user_shares returns a <flask_sqlalchemy.BaseQuery> object
user_shares = get_user_shares(user).with_entities(Share.id).subquery()
shares_roots = ShareRoot.query.filter(ShareRoot.sid.in_(user_shares)).subquery()
```
```python
SAWarning: Coercing Subquery object into a select() for use in IN(); please pass a select() construct explicitly shares_roots = ShareRoot.query.filter(ShareRoot.sid.in_(user_shares)).subquery()
```

**Comment by CaselIT** [comment, marked as answer]
Hi,

the warning is just saying that you should not pass a subquery to an `in_`, but a select.
So I think removing `.subquery()` from `user_shares ` should work (not sure since it's a flask object). Could you try?

**Reply by johaven** [reply]
i thinked i had to use the select statement but indeed it's working nice without subquery(). Thank you !

**Reply by CaselIT** [reply]
Query is translated to a select by sqlalchemy so it's considered select-like

**Reply by johaven** [reply]
Just a question, Query will be removed with SqlAlchemy 2.0 ?

**Reply by zzzeek** [reply]
Query is staying likely forever, no worries.  some parts of it like older join() syntaxes and the with_polymorphic() method will be removed, however.

**Reply by CaselIT** [reply]
the current plan like mike says is to mark it legacy, and keep it that way indefinitely with feature freeze. New development and project are encouraged to use select
